File: elucidate-common-lib/src/test/java/com/digirati/elucidate/common/infrastructure/util/PaginationUtilsTest.java
package com.digirati.elucidate.common.infrastructure.util;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class PaginationUtilsTest {

    @Test
    public void testLastPage() {
        assertEquals(PaginationUtils.calculateLastPage(0, 100), 0);
        assertEquals(PaginationUtils.calculateLastPage(99, 100), 0);
        assertEquals(PaginationUtils.calculateLastPage(100, 100), 0);
        assertEquals(PaginationUtils.calculateLastPage(101, 100), 1);
        assertEquals(PaginationUtils.calculateLastPage(199, 100), 1);
        assertEquals(PaginationUtils.calculateLastPage(200, 100), 1);
        assertEquals(PaginationUtils.calculateLastPage(201, 100), 2);
    }

}

File: elucidate-common-lib/src/test/kotlin/com/digirati/elucidate/common/infrastructure/util/PaginationUtilsTest.kt
package com.digirati.elucidate.common.infrastructure.util

import com.digirati.elucidate.common.infrastructure.util.PaginationUtils.calculateLastPage
import org.junit.Assert.assertEquals
import org.junit.Test

class PaginationUtilsTest {
    @Test
    fun testLastPage() {
        assertEquals(calculateLastPage(0, 100), 0)
        assertEquals(calculateLastPage(99, 100), 0)
        assertEquals(calculateLastPage(100, 100), 0)
        assertEquals(calculateLastPage(101, 100), 1)
        assertEquals(calculateLastPage(199, 100), 1)
        assertEquals(calculateLastPage(200, 100), 1)
        assertEquals(calculateLastPage(201, 100), 2)
    }
}